File: 1408_StringMatching.js


/**
 * @param {string[]} words
 * @return {string[]}
 */
var stringMatching = function(words) 
{
    let result = [];
    let str = words.join(",");
    console.log(str);
    for(word of words) {
        if(str.indexOf(word) !== str.lastIndexOf(word)) {
            result.push(word);
        }
    }
    return result;  
};

let main = function() {
    let words = ["uexk","aeuexkf","wgxih","yuexk","gxea","yuexkm","ypmfx","jjuexkmb","wqpri","aeuexkfpo","kqtnz","pkqtnza","nrbb","hmypmfx","krqs","jjuexkmbyt","zvru","ypmfxj"]
    console.log(stringMatching(words));
}
    
main();

File: 1408_StringMatching.cpp
#include<iostream>
#include<vector>
using namespace std;

class Solution 
{
    public:
        vector<string> stringMatching(vector<string>& words) 
        {
            vector<string> res;

            for(int i = 0; i < words.size(); i++)   
            {
                string word = words[i];
                for(int j = 0; j < words.size(); j++)
                {
                    if(i != j && words[j].size() >= words[i].size() && words[j].find(words[i]) != string::npos) {
                        res.push_back(words[i]);
                        break;
                    }
                }
            }
            return res;
        }
};

int main()
{
    return 0;
}